### PR TITLE
Fix: replaces `utf8_decode()` which is deprecated in PHP 8.2

### LINKED ISSUE #556: PHP 8.2 encoding deprecated functions
> PHP Deprecated:  mb_convert_encoding(): Handling HTML entities via mbstring is deprecated; use htmlspecialchars, htmlentities, or mb_encode_numericentity/mb_decode_numericentity instead in /woocommerce-pdf-invoices-packing-slips/includes/documents/abstract-wcpdf-order-document.php on line 899

> PHP Deprecated:  Function utf8_decode() is deprecated in /woocommerce-pdf-invoices-packing-slips/includes/documents/abstract-wcpdf-order-document.php on line 899

### PR DESCRIPTION
**Pull request by alexmigf**
closes #556

I have followed this solution: https://github.com/FriendsOfREDAXO/mblock/pull/156/commits/ce53b41e3ff9014dd54ee5864f8548bfb317c039

I could have used this:

```php
htmlspecialchars_decode(utf8_decode(htmlentities($html, ENT_COMPAT, 'utf-8', false)));
```

but `utf8_decode` is also deprecated.

### CODE REVIEW

**Comment by MohamadNateqi** [comment]
@alexmigf I happen to find this [PR](https://github.com/FriendsOfREDAXO/mblock/pull/161).
Did you already see it?

**Comment by alexmigf** [comment]
@MohamadNateqi actually no, good finding, I will review the implementation again.

**Comment by alexmigf** [comment]
@MohamadNateqi I have updated the code. I'm afraid [this](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/557/files#diff-8c34461e4a985e868c268f68c61cfb7378cd8c825fd007ed9a5d56468e6b657dR306) will only work when we merge the autoloader code https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/548

File: includes/wcpdf-functions.php [hunk 1]
```diff
@@ -291,3 +291,41 @@ function wcpdf_catch_db_object_errors( $wpdb ) {
...
+	}
+	
+	// 'utf8_decode()' is deprecated in PHP 8.2 and set to be removed on PHP 9.0.
+	if ( version_compare( PHP_VERSION, '8.2', '<' ) && function_exists( 'utf8_decode' ) ) {
+		return utf8_decode( $string );
+	}
+
+	// preferred fallback method for PHP 8.2+
+	if ( version_compare( PHP_VERSION, '8.1', '>' ) && class_exists( 'UConverter' ) ) {
+		return UConverter::transcode( $string, 'ISO-8859-1', 'UTF8' );
```
Comment (MohamadNateqi) [inline includes/wcpdf-functions.php:317]: Is it `UTF-8` or `UTF8`? Or is there no difference?

File: includes/wcpdf-functions.php [hunk 2]
```diff
@@ -291,3 +291,41 @@ function wcpdf_catch_db_object_errors( $wpdb ) {
...
+	if ( class_exists( '\\Symfony\\Polyfill\\Iconv\\Iconv' ) ) {
+		return \Symfony\Polyfill\Iconv\Iconv::iconv( 'UTF-8', 'ISO-8859-1', $string );
+	}
+
+	// default server library.
+	// PHP must have 'libiconv' configured instead of 'glibc' library.
+	if ( function_exists( 'iconv' ) ) {
+		return iconv( 'UTF-8', 'ISO-8859-1', $string );
+	}
+}
```
Comment (MohamadNateqi) [inline includes/wcpdf-functions.php:330]: Returning the string may be good if none of the available conversion methods are present.

Comment (alexmigf) [reply includes/wcpdf-functions.php:330]: It's unlikely but I will add.

**Comment by alexmigf** [comment]
> @alexmigf I think it's ok to merge it after the #548, yes?

Yes, we should wait the other to be merged first.

**Pull request merged**